From 5dae197d0f13357158f5ee4a61102604c21923c8 Mon Sep 17 00:00:00 2001
From: maxwen <max.weninger@gmail.com>
Date: Tue, 9 Oct 2018 13:52:55 +0200
Subject: [PATCH 03/12] camera: Allow to use boottime as timestamp reference

* Some /mad/ HALs use boottime clock without reporting
  timestamp source as realtime
  -> Add a flag to force boottime offset calculation

Change-Id: I56b623a1c2b58ca8a6287783d938fb665de201df
---
 services/camera/libcameraservice/Android.mk                | 4 ++++
 services/camera/libcameraservice/device3/Camera3Device.cpp | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/services/camera/libcameraservice/Android.mk b/services/camera/libcameraservice/Android.mk
index 0983de04b..2b3096a55 100644
--- a/services/camera/libcameraservice/Android.mk
+++ b/services/camera/libcameraservice/Android.mk
@@ -89,6 +89,10 @@ LOCAL_SHARED_LIBRARIES += \
     vendor.qti.hardware.camera.device@1.0
 endif
 
+ifeq ($(TARGET_CAMERA_BOOTTIME_TIMESTAMP),true)
+LOCAL_CFLAGS += -DTARGET_CAMERA_BOOTTIME_TIMESTAMP
+endif
+
 LOCAL_EXPORT_SHARED_LIBRARY_HEADERS := libbinder libcamera_client libfmq
 
 LOCAL_C_INCLUDES += \
diff --git a/services/camera/libcameraservice/device3/Camera3Device.cpp b/services/camera/libcameraservice/device3/Camera3Device.cpp
index 3409b32ce..e19097286 100644
--- a/services/camera/libcameraservice/device3/Camera3Device.cpp
+++ b/services/camera/libcameraservice/device3/Camera3Device.cpp
@@ -243,6 +243,10 @@ status_t Camera3Device::initializeCommonLocked() {
             ANDROID_SENSOR_INFO_TIMESTAMP_SOURCE_REALTIME) {
         mTimestampOffset = getMonoToBoottimeOffset();
     }
+#ifdef TARGET_CAMERA_BOOTTIME_TIMESTAMP
+    // Always calculate the offset if requested
+    mTimestampOffset = getMonoToBoottimeOffset();
+#endif
 
     // Will the HAL be sending in early partial result metadata?
     camera_metadata_entry partialResultsCount =
-- 
2.17.1

